SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -176,6 +169,8 @@ RunManagerMTWorker::RunManagerMTWorker(const edm::ParameterSet& iConfig, edm::Co
   if (m_LHCTransport) {
     m_LHCToken = iC.consumes<edm::HepMCProduct>(edm::InputTag("LHCTransport"));
   }
+  m_MagField = iC.esConsumes<MagneticField, IdealMagneticFieldRecord, edm::Transition::BeginRun>();
```

Comment:
Shouldn't this only be set if `m_pUseMagneticField == true`?

Reply:
It may be but this is not dangerous I guess. I will protect.

Reply:
I agree it isn't dangerous but without it the framework will have to do more work (i.e. prefetch the magnetic field even though you do not use it).

Reply:
Done